### PR TITLE
Add recovery endpoint

### PR DESCRIPTION
**Pull request by southeo**
Adds endpoint that accepts a list of handles and creates a DOI through datacite. This intended to fix DOI issues post-ingestion.

### CODE REVIEW

File: src/main/java/eu/dissco/core/datacitepublisher/security/WebSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@Configuration
+@EnableWebSecurity
+public class WebSecurityConfig  {
+
+  private final JwtAuthConverter jwtAuthConverter;
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http.authorizeHttpRequests(authorizeHttpRequests -> authorizeHttpRequests
+        .anyRequest().authenticated());
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/datacitepublisher/security/WebSecurityConfig.java]: I feel like authenticated would be enough as anyone can authenticate. Maybe we put this behind a role, maybe the `orchestration-admin` or see if we can create a generic `admin` role

File: src/main/java/eu/dissco/core/datacitepublisher/web/HandleClient.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+@RequiredArgsConstructor
+@Component
+@Slf4j
+public class HandleClient {
+
+  @Qualifier(value = "handle")
+  private final WebClient webClient;
+
+  public JsonNode resolveHandles(List<String> handles) throws HandleResolutionException {
```
Comment (samleeflang) [inline src/main/java/eu/dissco/core/datacitepublisher/web/HandleClient.java:25]: What if this list get's very big, will it still be able to handle that or should we set a maximum on this list, or even resolve them one-by-one?

**Review by samleeflang — COMMENTED** [review]
Looks good, just one or two small things

File: src/main/java/eu/dissco/core/datacitepublisher/security/WebSecurityConfig.java (hunk 1 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/datacitepublisher/security/WebSecurityConfig.java]: i'll add the orchestration-admin role for now. i was thinking this endpoint would only be accessed by another machine, maybe the handle api, but we don't have a pipeline for that yet.

File: src/main/java/eu/dissco/core/datacitepublisher/web/HandleClient.java (hunk 2 above)
Comment (southeo) [reply src/main/java/eu/dissco/core/datacitepublisher/web/HandleClient.java:25]: right, the responses can actually get quite large, we saw that with the specimen processor. Added some pagination here

**Review by samleeflang — APPROVED** [review]
:paintbrush:

**Pull request merged**